src/components/Todo/Todo.jsx

```diff
@@ -0,0 +1,209 @@
+import { useState, useContext } from "react";
+import { v4 as uuidv4 } from "uuid";
+import TextArea from "../TextArea";
+import Icon from "../Icon";
+import Tag from "../Tag";
+import Button from "../Button";
+import { format } from "date-fns";
+import { supabase } from "../../config/apiClient";
+import { AppContext } from "../../App";
+import classNames from "classnames";
+const Todo = ({ todo }) => {
+  const {
+    handleRemoveTodo,
+    flag,
+    removeCompleteFromIncomplete,
+    toasts,
+    setToasts,
+    search,
+  } = useContext(AppContext);
+  const [newName, setNewName] = useState(todo.name);
+  const [showEdit, setShowEdit] = useState(false);
+  const [showLoading, setShowLoading] = useState(false);
+  let completedDays = Math.ceil(
+    Math.abs(Date.parse(todo.completed_on) - Date.parse(todo.created_at)) /
+      (1000 * 60 * 60 * 24),
+  );
+  const editToggle = (e) => {
+    setShowEdit((prev) => !prev);
+  };
+  const edit = (e) => {
+    setNewName(e.target.value);
+  };
+  // Delete task
+  const deletetodo = (id) => {
+    setShowLoading(true);
+
+    handleRemoveTodo(id);
+  };
+  // on Complete
+
+  const completeHandler = async (e) => {
+    setShowLoading(true);
+    const dateValue = new Date(Date.now());
+
+    const { data, error } = await supabase
+      .from("ReactTodo")
+      .update({ completed_on: dateValue })
+      .match({ id: todo.id });
+
+    if (error === null) {
+      todo.completed_on = data[0]["completed_on"];
+      if (flag === "incomplete") removeCompleteFromIncomplete(todo.id);
+      if (showEdit) editToggle();
+    }
+    let newToast = {
+      id: uuidv4(),
+      type: error ? "error" : "success",
+      message: error ? error.message : "Task Complted",
+    };
+
+    setToasts([...toasts, newToast]);
+
+    if (flag !== "incomplete") setShowLoading(false);
+  };
+
+  const editValue = async (e) => {
+    if (
+      newName
+        .trim()
+        .replace(/\s+/g, " ")
```

Comment:
since we are reusing these, maybe we can store these in consts

Reply:
stored in the const and reused